pyth-rng/src/config.rs

```diff
@@ -90,3 +95,60 @@ pub struct RandomnessOptions {
     #[arg(default_value = "32")]
     pub chain_length: u64,
 }
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct Config {
+    pub chains: Vec<EthereumConfig>,
+}
+
+impl Config {
+    pub fn check_is_valid(&self) -> Result<(), Box<dyn Error>> {
+        let counts: HashMap<_, _> = self.chains.iter().fold(HashMap::new(), |mut map, s| {
+            *map.entry(s.chain_id.clone()).or_insert(0) += 1;
+            map
+        });
+
+        let duplicates: Vec<_> = counts
+            .iter()
+            .filter(|&(_, &c)| c > 1)
+            .map(|(id, _)| id.clone())
+            .collect();
+
+        if duplicates.len() > 0 {
+            Err(anyhow!(format!(
+                "Config has duplicated chain ids: {}",
+                duplicates.join(",")
+            ))
+            .into())
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn get_chain_config(&self, chain_id: &ChainId) -> Result<EthereumConfig, Box<dyn Error>> {
+        self.chains
+            .iter()
+            .find(|x| x.chain_id == *chain_id)
+            .map(|c| c.clone())
+            .ok_or(
+                anyhow!(format!(
+                    "Could not find chain id {} in the configuration file",
+                    &chain_id
+                ))
+                .into(),
+            )
+    }
```

Comment:
This method can go with the above suggestion, as you can just do `config.chains.get(chain_id)?` thanks to having a hashmap.

Reply:
i kept this because the error wrapping is useful (reduces code duplication because this is called in a few places). but implementation is much simpler with the hashmap in the config object